lib/liquid/c.rb

```diff
@@ -93,3 +83,41 @@ def parse(markup)
     end
   end
 end
+
+Liquid::Context.class_eval do
+  alias_method :ruby_evaluate, :evaluate
+  alias_method :ruby_find_variable, :find_variable
+
+  # This isn't entered often by Ruby (most calls stay in C via VariableLookup#evaluate)
+  # so the wrapper method isn't costly.
+  def c_find_variable_kwarg(key, raise_on_not_found: true)
```

Comment:
why is this needed? can't you define keyword argument Ruby methods in C?

Reply:
You can, but it's less efficient and a heck of a lot more code. From https://github.com/ruby/ruby/blob/v2_6_2/doc/extension.rdoc:

> Be warned, handling keyword arguments in the C API is less efficient than handling them in Ruby. Consider using a Ruby wrapper method around a non-keyword C function. ref: bugs.ruby-lang.org/issues/11339

Reply:
Put another way: either way we end up with 2 methods, since we wouldn't want to pass keyword arguments when calling C to C, so I opted to write the 2nd method in Ruby for the reasons mentioned.